### PR TITLE
AC-1961 Fetch presigned S3 URL for hero_image

### PR DESCRIPTION
**Pull request by rowellx68**
### Description

AC-1961 Fetch presigned S3 URL for hero_image

### Status
**Ready**

### Types of changes

- [ ] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New unit tests added to cover the changes.
- [ ] New e2e tests added to cover the changes. (If it has been decided these will not be added with the PR, a seperate ticket **must** be created and linked in the ticket below before merge)
- [ ] All tests passed locally.

### CODE REVIEW

File: src/modules/store/applications/applications.service.ts [hunk 1]
```diff
@@ -48,14 +53,28 @@ export class ApplicationsService {
...
+
+        const promises = data.applications.map(async (app) => {
+          app.hero_image = await this.s3Client.generatePresignedGetUrl(
+            app.hero_image,
+          );
+
+          return app;
+        });
+
+        data.applications = await Promise.all(promises);
```
Comment (rowellx68) [inline src/modules/store/applications/applications.service.ts:75]: @lillie-dae this is how I'm patching the `hero_image` value with the presigned URL

File: package.json [hunk 2]
```diff
@@ -22,11 +22,13 @@
...
   },
   "dependencies": {
+    "@aws-sdk/client-s3": "^3.301.0",
     "@aws-sdk/client-ses": "^3.290.0",
+    "@aws-sdk/s3-request-presigner": "^3.301.0",
     "@elastic/elasticsearch": "^8.6.0",
     "@golevelup/ts-jest": "^0.3.3",
     "@keycloak/keycloak-admin-client": "^18.0.2",
-    "@nestjs/axios": "^0.1.0",
+    "@nestjs/axios": "^2.0.0",
```
[the remark this answers is not in the document]
Comment (rowellx68) [reply package.json:31]: None that I know of. All tests still pass. We missed upgrading to 1.0

File: src/common/envs/.env [hunk 3]
```diff
@@ -54,5 +54,6 @@ AWS_SECRET_ACCESS_KEY=""
 
 AWS_SESSION_TOKEN=""
 AWS_REGION="eu-west-2"
+AWS_BUCKET=""
 
 AWS_SES_SENDER_EMAIL_ADDRESS="no-reply@dev-aide-app-store.answerdigital.io"
```
[the remark this answers is not in the document]
Comment (rowellx68) [reply src/common/envs/.env:59]: this should only be for local .env file, we replace this value on octopus on deployment

**Review by DanielNuttall-AD — APPROVED** [review]

File: src/common/envs/.env (hunk 3 above)
[the remark this answers is not in the document]
Comment (mehranrashid) [reply src/common/envs/.env:59]: I've done it in mine don't worry :)

**Review by mehranrashid — APPROVED** [review]

**Pull request closed without merging**